include/matx/core/make_sparse_tensor.h

```diff
@@ -55,11 +55,23 @@ auto make_tensor_coo(ValTensor &val, CrdTensor &row, CrdTensor &col,
   static_assert(ValTensor::Rank() == 1 && CrdTensor::Rank() == 1);
   using VAL = typename ValTensor::value_type;
   using CRD = typename CrdTensor::value_type;
-  using POS = int; // no positions, although some forms use [0,nse]
+  using POS = index_t;
+  // Note that the COO API typically does not involve positions.
+  // However, under the formal DSL specifications, the top level
+  // compression should set up pos[0] = {0, nse}. This is done
+  // here, using the same memory space as the other data.
+  POS *ptr;
+  matxMemorySpace_t space = GetPointerKind(val.GetStorage().data());
```

Comment:
Do you need to check for errors here?

Reply:
do you mean the consistency of memory spaces? that is done inside the sparse tensor constructor itself, right? or some other sort of errors?

Reply:
I mean what if the space is not found or is an error like `MATX_INVALID_MEMORY`

Reply:
Ah, I could intercept, but the allocator already has a MATX_THROW when the space is invalid right after that. Can we simply rely on that?